File: ccs/frontend/src/types/small_box_info.ts
/**
 * TypeScript 类型定义
 * 自动生成由 ccs-generate
 */

export interface SmallBoxInfo {
  id: number;
  small_box_number: number;
  parent_small_box_number: number | null;
  new_system_flag: boolean;
  arrival_date: string | null;
  created_at: string;
  updated_at: string;
}

export interface SmallBoxInfoCreate {
  small_box_number: number;
  parent_small_box_number: number | null;
  new_system_flag: boolean;
  arrival_date: string | null;
}

export interface SmallBoxInfoUpdate {
  small_box_number?: number;
  parent_small_box_number?: number | null;
  new_system_flag?: boolean;
  arrival_date?: string | null;
}

export interface ListResponse<T> {
  items: T[];
  total: number;
}

File: ccs/backend/app/schemas/small_box_info.py
"""
Pydantic Schema
自动生成由 ccs-generate
"""

from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import date, datetime


class SmallBoxInfoBase(BaseModel):
    small_box_number: int
    parent_small_box_number: Optional[int] = None
    new_system_flag: bool
    arrival_date: Optional[date] = None


class SmallBoxInfoCreate(SmallBoxInfoBase):
    pass


class SmallBoxInfoUpdate(SmallBoxInfoBase):
    pass


class SmallBoxInfoResponse(SmallBoxInfoBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
